dev/flutter/todo/submissions/Dishal Kanishka/todo_application/lib/screens/to_do_body.dart

```diff
@@ -0,0 +1,175 @@
+import "dart:convert";
+import "dart:math";
+
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:todo_application/Class/todo.dart';
+import 'package:todo_application/Constants/colors.dart';
+import 'package:todo_application/screens/view_todo.dart';
+
+class Body extends StatefulWidget {
+  Body({Key? key}) : super(key: key);
+
+  @override
+  State<Body> createState() => _BodyState();
+}
+
+class _BodyState extends State<Body> {
+  late SharedPreferences sharedPreferences;
+
+  bool switchValue = false;
+
+  List todos = [];
+
+  @override
+  void initState() {
+    super.initState();
+    initialGetSavedData();
+  }
+
+  void initialGetSavedData() async {
+    sharedPreferences = await SharedPreferences.getInstance();
+    String stringTodo = sharedPreferences.getString('todo')!;
+    List todoList = jsonDecode(stringTodo);
+    for (var todo in todoList) {
+      setState(() {
+        todos.add(Todo(id: 1, title: '', description: '', status: false)
+            .fromJson(todo));
+      });
+    }
+  }
+
+  void saveTodo() {
+    List items = todos.map((e) => e.toJson()).toList();
+    sharedPreferences.setString('todo', jsonEncode(items));
+  }
+
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Theme.of(context).appBarTheme.backgroundColor,
+        title: const Text('To Do'),
+      ),
+      backgroundColor: Theme.of(context).canvasColor,
+      body: ListView.builder(
+          scrollDirection: Axis.vertical,
+          itemCount: todos.length,
+          itemBuilder: (BuildContext context, int index) {
+            return Card(
+              elevation: 8.0,
+              margin:
+                  const EdgeInsets.symmetric(horizontal: 10.0, vertical: 6.0),
+              child: InkWell(
+                onTap: () async {
+                  Todo t = await Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                          builder: (context) => TodoView(todo: todos[index])));
+                  if (t != null) {
+                    setState(() {
+                      todos[index] = t;
+                    });
+                    saveTodo();
+                  }
+                },
+                child: makeListTile(todos[index], index),
```

Comment:
The `makeListTileWidget` is better to be separated out as a component within a components folder rather than being in this file itself. Also the case of it should be upper camel case

Reply:
Turn into upper camel case